### PR TITLE
hyprshade: init v0.9.3

### PR DESCRIPTION
**Pull request by willswats**
## Description of changes

Adds [Hyprshade](https://github.com/loqusion/hyprshade), a Hyprland shade configuration tool.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Squash the commits according to contributing guidelines

**Comment by natsukium** [comment]
I think this is not a python library but an application.
It should be added under `pkgs/applications/window-managers/hyprwm`, not the python module.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -18380,6 +18380,12 @@
     githubId = 20464732;
     name = "Willi Butz";
   };
+  willswats = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:18383]: Split this in a separate commit

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -5672,6 +5672,8 @@ with pkgs;
     wlroots = wlroots_0_15;
   };
 
+  hyprshade = python311Packages.callPackage ../applications/window-managers/hyprwm/hyprshade { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:5675]: It needs precisely this version of Python?

Comment (willswats) [reply pkgs/top-level/all-packages.nix:5675]: I was getting errors otherwise. It's defined [here](https://github.com/loqusion/hyprshade/blob/main/pyproject.toml) under `[project]` there is `requires-python = ">=3.11,<4.0"`.

Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:5675]: Ok then.

File: pkgs/applications/window-managers/hyprwm/hyprshade/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+, fetchFromGitHub
+, pdm
+, pdm-backend
+, more-itertools
+, click
+}:
+
+buildPythonPackage rec {
+  pname = "hyprshade";
+  version = "v0.9.3";
```
Comment (AndersonTorres) [inline pkgs/applications/window-managers/hyprwm/hyprshade/default.nix:12]:
```suggestion
  version = "0.9.3";
```

File: pkgs/applications/window-managers/hyprwm/hyprshade/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "loqusion";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/applications/window-managers/hyprwm/hyprshade/default.nix:17]:
```suggestion
    repo = "hyprshade";
```

File: pkgs/applications/window-managers/hyprwm/hyprshade/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "refs/tags/${version}";
```
Comment (AndersonTorres) [inline pkgs/applications/window-managers/hyprwm/hyprshade/default.nix:18]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/applications/window-managers/hyprwm/hyprshade/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  propagatedBuildInputs = [ pdm pdm-backend more-itertools click ];
+
+  meta = with lib; {
+    homepage = "https://github.com/loqusion/hyprshade";
+    description = "Hyprland shade configuration tool";
+    license = licenses.mit;
+    maintainers = with maintainers; [ willswats ];
+  };
```
Comment (AndersonTorres) [inline pkgs/applications/window-managers/hyprwm/hyprshade/default.nix:29]:
```suggestion
    platforms = platforms.linux;
  };
```

Because Ofborg says it does not run in Apple machines.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Squash the commits.

File: pkgs/applications/window-managers/hyprwm/hyprshade/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "loqusion";
+    repo = "hyprshade";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ou072V9nZUqf5DEolkMQy979SjaZs4iOuoszw50k4Y8=";
+  };
+
+  propagatedBuildInputs = [ pdm pdm-backend more-itertools click ];
```
Comment (natsukium) [inline pkgs/applications/window-managers/hyprwm/hyprshade/default.nix:22]:
```suggestion
  nativeBuildInputs = [
    pdm-backend
  ];

  propagatedBuildInputs = [ more-itertools click ];
```
`pdm-backend` is a build time only dependency.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Cleanup the second commit message:

```
hyprshade: init at 0.9.3

Co-authored-by: OTABI Tomoya <tomoya.otabi@gmail.com>
```

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
> ./results/hyprshade/bin/hyprshade ls
> Error: Command '['hyprctl', '-j', 'getoption', 'decoration:screen_shader']' returned non-zero exit status 1.

Is this expected?

**Comment by natsukium** [comment]
hyprctl is a Hyprland command.
Those who use this tool should use Hyprland, so I don't think it's a problem.
In fact, it worked on my Hyprland environment.

**Review by natsukium — APPROVED** [review]
LGTM

**Comment by AndersonTorres** [comment]
Usually it implies some form of soft dependency. The program runs OK but it is not able to do anything useful.
The typical example are plugins for mpv.

But I will not pursue this further.
